README.md

```diff
@@ -15,7 +94,53 @@ have pre-slotted error codes for each library. The currently allocated error ran
 | [0, 0x0400) | aws-c-common |
 | [0x0400, 0x0800) | aws-c-io |
 | [0x0800, 0x1000) | aws-c-http |
-| [0x1000, 0x2000) |aws-c-amazon-flow | 
+| [0x1000, 0x2000) |aws-c-eventstream | 
 | [0x2000, 0x2800) | (reserved for future project) |
 
-Each library should begin its error codes at the beginning of its range and follow in sequence (don't skip codes).
+Each library should begin its error codes at the beginning of its range and follow in sequence (don't skip codes). Upon
+adding an AWS maintained library, an error code range must be approved and added to the above table.
+
+### Testing
+We have a high bar for test coverage, and PRs fixing bugs or introducing new functionality need to have tests before 
+they will be accepted. A couple of tips:
+
+#### Aws Test Harness
+We provide a test harness for writing unit tests. This includes an allocator that will fail your test if you have any 
+memory leaks, as well as some `ASSERT` macros. To write a test:
+
+* Create a *.c test file in the tests directory of the project. 
+* Implement one or more tests with the signature `int test_case_name(struct aws_allocator *, void *ctx)`
+* Use the `AWS_TEST_CASE` macro to declare the test.
+* Include your test in the `tests/main.c` file.
+* Include yur test in the `tests/CMakeLists.txt` file. 
+
+### Coding Style
+* No Tabs
+* Indent is 4 spaces
+* K & R style for braces
+* Space after if, before the `(`
+* Avoid C99 features in header files
+* Avoid C++ style comments e.g. `//`
+* All public API functions need C++ guards and Windows dll semantics
+* Use Unix line endings
+* SNAKE_UPPER_CASE constants, macros, and enum members.
+* snake_lower_case everything else.
+* do not use typedef struct idiom.
+* typedef function pointer definitions if you intend to expose them to the user
+* typedef enums
+* every source and header file must have a copyright header (The standard AWS one for apache 2).
+* Use standard include guards (e.g. #IFNDEF HEADER_NAME #define HEADER_NAME etc...)
```

Comment:
I'm personally in favor of using "#pragma once" because otherwise we can easily run into issues such as the current include/aws/common/byte_buf.h file having an include guard of AWS_STRING_H due to changed filenames. If we don't maintain those include guards well, we can easily end up breaking things due to include guard naming collisions. Also "#pragma once" is one line of code, instead of three for the include guard.

Reply:
#pragma once expects a C99 language standard, which we don't want to expose in a header file, for non C99 programs.

I also "prefer" it, but I'd rather not hamstring a user without a really good reason. (ergonomics isn't a good enough reason).

Reply:
It works fine for me to compile code with #pragma once and -std=c89 using gcc/clang on both my Linux dev desktop and my Mac laptop. I even used a redundant include to make sure that #pragma once was doing what it was supposed to be doing:

(18-04-13 16:34:14) <0> [~/sandbox]
 % cat once.h
#pragma once

struct foo {
    int x, y;
};

(18-04-13 16:34:24) <0> [~/sandbox]
 % cat once.c
#include "once.h"
#include "once.h"
#include <stdio.h>

int main() {
    struct foo f = {5, 7};
    printf("%i\t%i\n", f.x, f.y);
    return 0;
}

(18-04-13 16:34:25) <0> [~/sandbox]
 % gcc -Wall -Werror -pedantic -std=c89 -o once once.c

(18-04-13 16:34:31) <0> [~/sandbox]
 % ./once
5	7

That was on Linux. Results on Mac were the same. If there really is a portability problem with #pragma once on other systems/compilers, then I agree with your reasoning.

Reply:
the main goal here is that we want to support this from any compiler (even ancient ones using ANSI C), adding #pragma once is similar to using C99 syntax in a header file.